CMSgov/ato-blueprint
# create new project model and initial set up

https://jiraent.cms.gov/browse/ISPGBSS-755

### Changes
- Added a project directory to house a model, view, urls file for project
- Created a new project model which is temporarily called `package` to limit interference with the existing project model. _Note: the rename should happen when the former project model is removed._
    - This created two tables in the database: `project.package` and `project_package_components`. The latter was automatically created to handle the "many to many relationship" between projects and components
- Added urls for project pages as placeholders (per [this spreadsheet](https://docs.google.com/spreadsheets/d/18zvZJb3mQf3f1OHGLVbR6JrRV62rdEL_BporHc_QXwc/edit?pli=1#gid=0))
- Added a view file to reference the existing project.html. Some refactor work will be needed to the project.html template and view in order to further use the new model

### Testing
- To populate the data, add a row to the new project model in the `project_package` table of the database. Here is a sample command:
   - _You may need to change the value for `creator_id` from "2" to a corresponding value in your database._
```
INSERT INTO project_package(title, acronym, created, creator_id)
VALUES ('Testing Tapir', 'TT', current_timestamp, 2)
```
- Ensure there are no errors when inserting data and run a query on the `project_package` table to verify the insert.
- Check the ability to update the row with additional field data. Sample command:
    - _You **will** need to change the value for `id` from "9" to the id that was assigned to your INSERT INTO statement above._
    - _You may need to change the value for `catalog_id` from "1" to a corresponding value in your database._
```
UPDATE project_package
SET
    catalog_id = 1
    impact_level = 'moderate',
    location = 'cms_aws’,
    updated = current_timestamp
WHERE
    id = 9
```
- Ensure there are no errors when updating data and run a query on the `project_package` table to verify the update.
- Check that we can update the `project_package_components` table (this table handles the “many to many relationship” between projects and components). Sample command:
   - _You **will** need to change the values for `package_id` from "6", "7", "8" to corresponding values in your database._
   - _You **will** need to change the values for `element_id` from "34", "35" to corresponding values in your database._
```
INSERT INTO project_package_components(package_id, element_id)
VALUES
  (6, 34),
  (6, 35),
  (7, 34),
  (8, 34)
```
- Ensure there are no errors when inserting data and run a query on the `project_package_components` table to verify the insert.
- Visit `/project/<project_id>` and see that it loads the page using the project.html template.

### Developer Notes
Future work will be needed for:
- Inserting a new project into the new project data model
- Refactoring the project.html template and view in order to further use the new model
- Cleaning up: removing the old project model and related code, renaming the new project model from "Package" to "Project"
- Create new a new model for components (related work, but not necessarily dependent)
